debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlReadOnlyIncrementalSnapshotChangeEventSource.java

```diff
@@ -92,20 +91,27 @@ public MySqlReadOnlyIncrementalSnapshotChangeEventSource(RelationalDatabaseConne
                                                              DataChangeEventListener<MySqlPartition> dataChangeEventListener,
                                                              NotificationService<MySqlPartition, MySqlOffsetContext> notificationService) {
         super(config, jdbcConnection, dispatcher, databaseSchema, clock, progressListener, dataChangeEventListener, notificationService);
-        kafkaSignal = new KafkaSignalChannel();
     }
 
     @Override
     public void init(MySqlPartition partition, OffsetContext offsetContext) {
         super.init(partition, offsetContext);
 
-        kafkaSignal.init(connectorConfig);
-        Long signalOffset = getContext().getSignalOffset();
-        if (signalOffset != null) {
-            kafkaSignal.seek(signalOffset);
+        if (isKafkaChannelEnabled()) {
+
+            restoreOffset();
         }
     }
 
+    private void restoreOffset() {
+
+        dispatcher.getSignalProcessor().restoreKafkaOffset(getContext().getSignalOffset());
```

Comment:
Instead of adding `restoreKafkaOffset` to `SignalProcessor` would it be possible to extend `SignalChannelReader` with a method with a generic type that could be overridden by implementations. In case of `KafkaSignalChannel` it will be type `long`.
This method would just ask for `kafka` channel to be provided by `SignalProcessor`. If available the method will be called and ignored otherwise.

Reply:
@jpechane Can you please better explain this sentence?
> This method would just ask for kafka channel to be provided by SignalProcessor. If available the method will be called and ignored otherwise.

Reply:
@mfvitale Sorry, it will call `getSignalChannel` to get the channel instance and call the method on it.

Reply:
So if I understood well your proposal is to:

- add a method in `SignalChannelReader` let's call it `restoreOffset(T offset)` maybe with an empty default implementation
- implement this method in the  `KafkaSignalChannel` with Long type
- ySqlReadOnlyIncrementalSnapshotChangeEventSource will call getSignalChannel and the call the the method for restoring the offset

Reply:
Wow `MySqlReadOnlyIncrementalSnapshotChangeEventSource` is a very long class name. Up for Scrabble anyone?

Reply:
@mfvitale Exactly, just I'd like to use different name, like `init`/`reset` or similar to make it generic and re-usable if needed elswhere.

Reply:
Even if this is not strictly related to this PR, honestly I don't think it's to long. It tells exactly what it is. Do you think there is a better name?